xcube/core/store/store.py

```diff
@@ -690,3 +691,22 @@ def deregister_data(self, data_id: str):
         Raises:
             DataStoreError: If an error occurs.
         """
+
+
+class PreloadDataStore(MutableDataStore):
+    """A preload data store is a multable data store which contains the preload handle.
+
+    Instances of this class are returned by the ``DataStore.preload_data()`` method.
+    """
+
+    @property
+    @abstractmethod
+    def preload_handle(self) -> PreloadHandle:
+        """This field must be defined and set in subclasses."""
+        pass
+
+    @preload_handle.setter
+    @abstractmethod
+    def preload_handle(self, value: PreloadHandle) -> None:
+        """Setter for preload_handle."""
+        pass
```

Comment:
This should not be part of the interface. Users should not change the handle. The handle is set by data store implementors, not users.